### PR TITLE
feat: auto-fetch latest OSTK core patch version in docker image

### PR DESCRIPTION
**Pull request by kyle-cochran**
ADDs a json file that comes from the Github refs/tags api. We can query to match a semver pattern. For example:
`https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/matching-refs/tags/0.7`
Returns all tags matching `0.7`. i.e., if we follow semver properly, all patch tags. This returns:
```
[
  {
    "ref": "refs/tags/0.7.0",
    "node_id": "MDM6UmVmMTM2ODM0NzAwOnJlZnMvdGFncy8wLjcuMA==",
    "url": "https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/refs/tags/0.7.0",
    "object": {
      "sha": "3b378692446add8d51251df34a954f23d0374623",
      "type": "commit",
      "url": "https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/commits/3b378692446add8d51251df34a954f23d0374623"
    }
  },
  {
    "ref": "refs/tags/0.7.1",
    "node_id": "MDM6UmVmMTM2ODM0NzAwOnJlZnMvdGFncy8wLjcuMQ==",
    "url": "https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/refs/tags/0.7.1",
    "object": {
      "sha": "58f780a9921162dc451c5d4edd0b149b2f9d5223",
      "type": "commit",
      "url": "https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/commits/58f780a9921162dc451c5d4edd0b149b2f9d5223"
    }
  },
  {
    "ref": "refs/tags/0.7.2",
    "node_id": "MDM6UmVmMTM2ODM0NzAwOnJlZnMvdGFncy8wLjcuMg==",
    "url": "https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/refs/tags/0.7.2",
    "object": {
      "sha": "c370cf41622e3f244c88649bbac0f69588ba65c1",
      "type": "commit",
      "url": "https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/commits/c370cf41622e3f244c88649bbac0f69588ba65c1"
    }
  }
]
```

We can then use `jq` to extract the `ref` field of the last element, which is always the latest version:
`jq -r .[-1].ref versions.json`

and cut away the `refs/tags/` part:
`cut -d "/" -f3`

### CODE REVIEW

File: docker/development/Dockerfile [hunk 1]
```diff
@@ -48,12 +53,17 @@ RUN git clone --branch ${GEOMETRIC_TOOLS_ENGINE_VERSION} --depth 1 https://githu
...
+## Force an image rebuild when new Core patch versions are detected
+ADD https://api.github.com/repos/open-space-collective/open-space-toolkit-core/git/matching-refs/tags/${OSTK_CORE_MAJOR}.${OSTK_CORE_MINOR} /tmp/open-space-toolkit-core/versions.json
 
 RUN mkdir -p /tmp/open-space-toolkit-core \
  && cd /tmp/open-space-toolkit-core \
- && wget --quiet https://github.com/open-space-collective/open-space-toolkit-core/releases/download/${OSTK_CORE_VERSION}/open-space-toolkit-core-${OSTK_CORE_VERSION}-1.x86_64-runtime.deb \
- && wget --quiet https://github.com/open-space-collective/open-space-toolkit-core/releases/download/${OSTK_CORE_VERSION}/open-space-toolkit-core-${OSTK_CORE_VERSION}-1.x86_64-devel.deb \
+ && export LATEST_PATCH_VERSION=$(jq -r .[-1].ref versions.json | cut -d "/" -f3) \
+ && wget --quiet https://github.com/open-space-collective/open-space-toolkit-core/releases/download/${LATEST_PATCH_VERSION}/open-space-toolkit-core-${LATEST_PATCH_VERSION}-1.x86_64-runtime.deb \
+ && wget --quiet https://github.com/open-space-collective/open-space-toolkit-core/releases/download/${LATEST_PATCH_VERSION}/open-space-toolkit-core-${LATEST_PATCH_VERSION}-1.x86_64-devel.deb \
```
Comment (vishwa2710) [inline docker/development/Dockerfile:66]:
```suggestion
 && export LIBRARY_VERSION=$(jq -r .[-1].ref versions.json | cut -d "/" -f3) \
 && wget --quiet https://github.com/open-space-collective/open-space-toolkit-core/releases/download/${LIBRARY_VERSION}/open-space-toolkit-core-${LIBRARY_VERSION}-1.x86_64-runtime.deb \
 && wget --quiet https://github.com/open-space-collective/open-space-toolkit-core/releases/download/${LIBRARY_VERSION}/open-space-toolkit-core-${LIBRARY_VERSION}-1.x86_64-devel.deb \
```
Nit: LATEST_PATCH_VERSION sounds like it's just the latest - patch version. i.e. .2 in the case of Core.

**Review by vishwa2710 — APPROVED** [review]
Sick

File: docker/development/Dockerfile (hunk 1 above)
Comment (kyle-cochran) [reply docker/development/Dockerfile:66]: Hmm... I see what you mean. I'd like to have it say something a bit more explicit in order to demystify the jq stuff. `LATEST_PATCH_OF_MINOR`?

Comment (vishwa2710) [reply docker/development/Dockerfile:66]: Actually never mind, latest patch version has grown on me. I like it.

**Review by vishwa2710 — APPROVED** [review]

**Review by vishwa2710 — APPROVED** [review]

**Pull request merged**